ddterm/app/application.js

```diff
@@ -659,14 +659,26 @@ class Application extends Gtk.Application {
                 false
             );
 
+            if (!data_variant.is_normal_form()) {
+                logError(new Error('Invalid session file data variant, delete session file.'));
+                GLib.unlink(this.session_file_path);
+                return;
+            }
+
             this.ensure_window().deserialize_state(data_variant);
+        } else {
```

Comment:
This `else` gets executed when the file is empty - not on error. And what would happen if something in `deserialize_state()` (or in functions/methods it calls) throws?

Reply:
removed the else